code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -360,12 +360,7 @@
 				to_chat(user, "<span class='warning'>You can't swap your gender!</span>")
 				return 5 SECONDS
 
-			if(user.gender == MALE)
-				user.gender = FEMALE
-				user.visible_message("<span class='boldnotice'>[user] suddenly looks more feminine!</span>", "<span class='boldwarning'>You suddenly feel more feminine!</span>")
-			else
-				user.gender = MALE
-				user.visible_message("<span class='boldnotice'>[user] suddenly looks more masculine!</span>", "<span class='boldwarning'>You suddenly feel more masculine!</span>")
+			user.set_gender(user.gender == MALE ? FEMALE : MALE, forced = TRUE) //You are doing this to yourself.
```

Comment:
Why remove the message that indicates something happened?

Reply:
Was moved to this proc inheritor.
https://github.com/BeeStation/BeeStation-Hornet/blob/ba1f45be26c47a18801337ffacb456856a7d607d/code/modules/mob/living/carbon/carbon.dm#L987